src/box/memtx_engine.c

```diff
@@ -381,6 +381,8 @@ memtx_engine_prepare(struct engine *engine, struct txn *txn)
 		if (stmt->add_story != NULL || stmt->del_story != NULL)
 			memtx_tx_history_prepare_stmt(stmt);
 	}
+	if (txn->is_schema_changed)
+		memtx_tx_abort_all_for_ddl(txn);
```

Comment:
We have [space_invalidate](https://github.com/tarantool/tarantool/blob/b1ee4e59b6450812fb68f34144d7ac7d639623bc/src/box/schema.cc#L259) callback, which is currently used by Vinyl to abort all pending transactions on DDL. Can we reuse it for memtx?